erdjs/src/message/message.ts

```diff
@@ -0,0 +1,78 @@
+import { sign } from "tweetnacl";
+import { Address } from "../address";
+import { ISignable } from "../interface";
+import { Signature } from "../signature";
+import { MessageVersion, MessageSigner } from "./messageParams";
+
+const MESSAGE_VERSION = new MessageVersion(1);
+
+export class Message implements ISignable {
```

Comment:
In case of message signing, to keep this compatible with other protocols and prevent an attacker to mislead someone into signing a transaction, the message to be signed needs to prepend the prefix "\x17Elrond Signed Message:\n" and the signing needs to be done on the resulting message (with this prefix).

Ledger binary already does this so you do not need to add the prefix when signing with ledger, but when signing from erdjs then you need to add it.

Reply:
Updated

Reply:
@alexandru-g also please make sure the flow follows the examples from here:
https://github.com/ElrondNetwork/elrond-go/blob/development/examples/messageSign_test.go

Reply:
Perfect, thanks, I'll update the code to sign a hash instead of the actual payload.

Reply:
Added signing on keccak256 hashed payload.